docs/api-reference/effects/glfx-shader-modules.md

```diff
@@ -0,0 +1,342 @@
+        <img height=340 src="https://raw.githubusercontent.com/uber-common/deck.gl-data/master/images/samples/glfx/results/vibrance.jpg" />
+        <p><i>Vibrance Effect</i></p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+
+### vignette
+
+Adds a simulated lens edge darkening effect.
+
+* `size`     0 to 1 (0 for center of frame, 1 for edge of frame). Default value is 0.5.
+* `amount`   0 to 1 (0 for no effect, 1 for maximum lens darkening). Default value is 0.5.
+
+<table style="border: 0;" align="center">
+  <tbody>
+    <tr>
+      <td align="center">
+        <img height=340 src="https://raw.githubusercontent.com/uber-common/deck.gl-data/master/images/samples/glfx/results/vignette.jpg" />
+        <p><i>Vignette Effect</i></p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+
+### tiltShift
+
+Simulates the shallow depth of field normally encountered in close-up photography, which makes the scene seem much smaller than it actually is. This filter assumes the scene is relatively planar, in which case the part of the scene that is completely in focus can be described by a line (the intersection of the focal plane and the scene). An example of a planar scene might be looking at a road from above at a downward angle. The image is then blurred with a blur radius that starts at zero on the line and increases further from the line.
+
+ * `start`          [x, y] coordinate of the start of the line segment in pixels. Default value is [0, 0] which is the bottom left corner.
+ * `end`            [x, y] coordinate of the end of the line segment in pixels. Default value is [1, 1].
+ * `blurRadius`     The maximum radius of the pyramid blur in pixels. Default value is 15.
+ * `gradientRadius` The distance in pixels from the line at which the maximum blur radius is reached. Default value is 200.
+
+<table style="border: 0;" align="center">
+  <tbody>
+    <tr>
+      <td align="center">
+        <img height=340 src="https://raw.githubusercontent.com/uber-common/deck.gl-data/master/images/samples/glfx/results/tilt_shift.jpg" />
+        <p><i>Tilt Shift Effect</i></p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+
+### triangleBlur
+
+This is the most basic blur filter, which convolves the image with a pyramid filter. The pyramid filter is separable and is applied as two perpendicular triangle filters.
+
+* `radius` The radius of the pyramid in pixels convolved with the image. Default value is 20.
+
+<table style="border: 0;" align="center">
+  <tbody>
+    <tr>
+      <td align="center">
+        <img height=340 src="https://raw.githubusercontent.com/uber-common/deck.gl-data/master/images/samples/glfx/results/triangle_blur.jpg" />
+        <p><i>Triangle Blur Effect</i></p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+
+### zoomBlur
+
+Blurs the image away from a certain point, which looks like radial motion blur.
+
+* `center`  [x, y] coordinate of the blur origin in pixels. Default value is [0, 0] which is the bottom left corner.
```

Comment:
Any chance we can make this coordinate relative to the screen size?

Consider this scenario in deck: how can I create a zoom blur effect that always center in the viewport? I have to listen to the `onResize` event of deck, and create a new `PostProcessEffect` every time, because there is no way for me to update the props of an existing one. This will lead to constant creation of new models.

Reply:
That's how the glfx example implemented, user doesn't resize frequently. We can add a screenSize uniform in all the shaders, and update examples. May not catch up the 7.1 release. Do we want to go this new solution?

Reply:
how about use texture size instead of screen size? in our case they are the same.

Reply:
> user doesn't resize frequently

This is not true. When the canvas resize is triggered by dragging the window or CSS animation, `onResize` would be called on every animation frame.